example/exampleContour.html

```diff
@@ -0,0 +1,210 @@
+<body>
+<nav class="navbar navbar-default">
+    <div class="container-fluid">
+        <div class="navbar-header" style="margin-top:10px">
+            <span class="heavyai">HEAVY.AI Demo</span>
+        </div>
+        <div class="navbar-text navbar-right">
+            <div class="data-count"></div>
+        </div>
+    </div>
+</nav>
+<div class="main-container">
+    <div>
+        <div class="title">Contour Map</div>
+        <div id="contourChart"></div>
+    </div>
+</div>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.13.0/moment.min.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/lodash.js/4.17.4/lodash.min.js"></script>
+<script src="https://code.jquery.com/jquery-2.2.4.min.js" integrity="sha256-BbhdlvQf/xTY9gja0Dq3HiwQF8LaCRTXxZKRutelT44=" crossorigin="anonymous"></script>
+<script src="assets/app.bundle.js"></script>
+
+<script>
+
+  function createCharts(crossFilter, dc, config, con) {
+    var w = Math.max(document.documentElement.clientWidth, window.innerWidth || 0)
+    var h = Math.max(document.documentElement.clientHeight, window.innerHeight || 0) - 100
+
+    // Linemap Chart
+    const parent = document.getElementById("contourChart");
+    const mapboxToken = "pk.eyJ1IjoibWFwZCIsImEiOiJjaWV1a3NqanYwajVsbmdtMDZzc2pneDVpIn0.cJnk8c2AxdNiRNZWtx5A9g";
+
+    const countGroup = crossFilter.groupAll();
+
+    dc.countWidget(".data-count")
+      .dimension(crossFilter)
+      .group(countGroup);
+
+    const ContourChart = dc.rasterChart(parent, true, null, mapboxgl)
+      .con(con)
+      .height(h)
+      .width(w)
+      .mapUpdateInterval(750)
+      .mapStyle('mapbox://styles/mapbox/light-v8')
+      .mapboxToken(mapboxToken) // need a mapbox accessToken for loading the tiles
+      .popupSearchRadius(2)
+      .useLonLat(true)
+      .useGeoTypes(true) // need for projecting geo column using "mercator_map_projection"
+
+
+    const contourLayer = dc.rasterLayer("lines")
+      .crossfilter(crossFilter)
+      .setState({
+        data: [{
+          type: "contour",
+          table: config.table,
+          source: config.table,
+          contour_value_field: config.value,
+          lat_field: config.lat_field,
+          lon_field: config.lon_field,
+          agg_type: 'AVG',
+          fill_agg_type: 'AVG',
+          bin_dim_meters: config.bin_dim_meters,
+          neighborhood_fill_radius: 0,
+          fill_only_nulls: false,
+          flip_latitude: false,
+          minor_contour_interval: config.minor_contour_interval,
+          major_contour_interval: config.major_contour_interval,
+          contour_offset: 0,
+        }],
+        transform: {
+          projection: "mercator_map_projection",
+        },
+        mark: {
+          type: "lines",
+          lineJoin: "bevel",
+        },
+        encoding: {
+          // Here so that lines layer doesn't break
+          size: "auto",
+          color: {
+          },
+          // Contour stuff
+          stroke: {
```

Comment:
curious, was the `stroke` property exposed/working before w/ scales? I'm guessing not. I don't think there's a current immerse case where a scale is used for stroke width (except for maybe th auto-sizing thing, which I can't claim to know how that works)

I think this is fine for now, but I think it would be worth revisiting this to avoid doing a property specialization inside the encoding section. For instance, I see you can attach a color to the major/minor here, but how does that work with the `color` property above? What happens if there's a `color` defined above, and a `color` here for major/minor?

I think we should instead look at something like this, where we define an attribute name/tag or major/minor in the `contour` data block, so something like this:

```
{
  ...,
  data: [
    {
      type: "contour",
      ...,
      intervals: {
        major: config.major_contour_interval,
        minor: config.minor_contour_interval,
        columnName: "is_major_contour" // output column name that tags major/minor
      },
      ...
    }
  ],
  ...,
  encoding: {
    ...,
    color: {
      field: "is_major_contour",
      type: "nominal",
      scale: {
        domain: [true, false],
        range: ["red", "blue"]
      }
    },
    strokeWidth: {
      field: "is_major_contour",
      type: "nominal",
      scale: {
        domain: [true, false],
        range: [3, 1]
      }
    },
    ...
  }
}
```

What I like about this approach is the `color` and `strokeWidth` properties have no specializations for major/minor, and going one step further, we could use styles to avoid redundancy.

However, you do lose some of the explicit-ness/readability of your json.

Reply:
Im going to revisit this when doing the contour polygon changes and leave it this way for now, but I agree with the suggestion and will shoot for a format similar to this on the next go